### PR TITLE
GG-28610 SQL: Improve exception handling in the case when query quota is exceeded

### PR DESCRIPTION
**Pull request by rkondakov**

### CODE REVIEW

File: modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/QueryMemoryManager.java [hunk 1]
```diff
@@ -212,8 +211,7 @@ public boolean onQuotaExceeded(long size) {
         if (offloadingEnabled)
             return false;
         else {
-            throw new IgniteSQLException("SQL query run out of memory: Global quota exceeded.",
-                IgniteQueryErrorCode.QUERY_OUT_OF_MEMORY);
+            throw new SqlMemoryQuotaExceededException("SQL query run out of memory: Global quota exceeded.");
```
Comment (MaxLosevskoy) [inline modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/QueryMemoryManager.java:214]: Typo in `...query ran out...`

File: modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/QueryMemoryTracker.java [hunk 2]
```diff
@@ -168,8 +167,7 @@ private boolean onQuotaExceeded() {
         if (offloadingEnabled)
             return false;
         else
-            throw new IgniteSQLException("SQL query run out of memory: Query quota exceeded.",
-                IgniteQueryErrorCode.QUERY_OUT_OF_MEMORY);
+            throw new SqlMemoryQuotaExceededException("SQL query run out of memory: Query quota exceeded.");
```
Comment (MaxLosevskoy) [inline modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/QueryMemoryTracker.java:170]: Typo in `ran`

File: modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/QueryMemoryManager.java (hunk 1 above)
Comment (rkondakov) [reply modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/QueryMemoryManager.java]: Fixed

File: modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/QueryMemoryTracker.java (hunk 2 above)
Comment (rkondakov) [reply modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/QueryMemoryTracker.java]: Fixed

**Pull request closed without merging**